aptos-move/move-examples/large_packages/sources/large_packages.move

```diff
@@ -34,42 +88,84 @@ module large_packages::large_packages {
-            let idx = (vector::pop_back(&mut code_indices) as u64);
-            while (vector::length(&staging_area.code) <= idx) {
-                vector::push_back(&mut staging_area.code, vector::empty());
+        if (!vector::is_empty(&metadata_chunk)) {
+            vector::append(&mut staging_area.metadata_serialized, metadata_chunk);
+        };
+
+        let i = 0;
+        while (i < vector::length(&code_chunks)) {
+            let inner_code = *vector::borrow(&code_chunks, i);
+            let idx = (*vector::borrow(&code_indices, i) as u64);
+
+            if (smart_table::contains(&staging_area.code, idx)) {
+                vector::append(smart_table::borrow_mut(&mut staging_area.code, idx), inner_code);
+            } else {
+                smart_table::add(&mut staging_area.code, idx, inner_code);
+                if (idx > staging_area.last_module_idx) {
+                    staging_area.last_module_idx = idx;
+                }
             };
-            let source_code = vector::borrow_mut(&mut staging_area.code, idx);
-            vector::append(source_code, inner_code)
+            i = i + 1;
         };
 
-        let _ = staging_area;
+        staging_area
+    }
 
-        if (publish) {
-            publish_staged_code(owner, owner_address);
-            move_from<StagingArea>(owner_address);
-        }
+    inline fun publish_to_account(
+        publisher: &signer,
+        staging_area: &mut StagingArea,
+    ) {
+        let code = assemble_module_code(staging_area);
+        code::publish_package_txn(publisher, staging_area.metadata_serialized, code);
     }
 
-    public entry fun cleanup(owner: &signer) acquires StagingArea {
-        move_from<StagingArea>(signer::address_of(owner));
+    inline fun publish_to_object(
+        publisher: &signer,
+        staging_area: &mut StagingArea,
+    ) {
+        let code = assemble_module_code(staging_area);
+        object_code_deployment::publish(publisher, staging_area.metadata_serialized, code);
     }
 
-    /// Publish code from staging area.
-    public entry fun publish_staged_code(
+    inline fun upgrade_object_code(
         publisher: &signer,
-        staging_area_address: address,
-    ) acquires StagingArea {
-        let staging_area = borrow_global_mut<StagingArea>(staging_area_address);
-        code::publish_package_txn(publisher, staging_area.metadata_serialized, staging_area.code);
+        staging_area: &mut StagingArea,
+        code_object: Object<PackageRegistry>,
+    ) {
+        let code = assemble_module_code(staging_area);
+        object_code_deployment::upgrade(publisher, staging_area.metadata_serialized, code, code_object);
+    }
+
+    inline fun assemble_module_code(
+        staging_area: &mut StagingArea,
+    ): vector<vector<u8>> {
+        let last_module_idx = staging_area.last_module_idx;
+        let code: vector<vector<u8>> = vector[];
+        let i: u64 = 0;
```

Comment:
nit: Do we need these type declarations? Can remove `u64` and `vector<vector<u8>>`